### DISCUSSION TITLE
Performance Probleme zu großer Datensatz

### DISCUSSION THREAD

**Discussion by LeonMunz** [irgroup/dis25-2021#19] [Q&A]
Hallo zusammen,
wir sind auf ein größeres Problem gestoßen. 
Wir haben unsere Cluster auf die unique suggestion terms berechnet und möchten den gesamten Datensatz aggregieren, um die Anzahl der Cluster zu bestimmen.
Allerdings schaffen unsere Rechner die Berechnung auf den gesamten Datensatz nicht. Bzw. haben wir sogar den EU-Datensatzes auf ein Jahr gefiltert und nochmal in Batches aufgeteilt.
Wir haben es mit Dask versucht allerdings bekommen wir selbst damit memory errors weswegen wir nun ein bisschen ratlos sind wie wir diese Menge Daten verarbeiten sollen.

Aktuell haben wir zwei Optionen im Kopf, um das Problem zu umgehen:
1)	Einen Server mieten, um die Berechnung auszulagern
2)	Das Problem in der Ausarbeitung zu beschreiben und die Berechnung nur auf einen verkürzten Datensatz auszuführen (Was die Ergebnisse natürlich verfälscht)

Wie sollen wir weiter vorgehen? Gibt es eine präferierte Option oder eine die wir nicht auf dem Schirm haben? 
Danke im Voraus und beste Grüße.

**Comment by phschaer** [comment]
Wenn es ein RAM-Problem ist… wir hätten in meiner Arbeitsgruppe einen Server mit potentiell 768 GByte RAM. Da müsste ich aber erstmal in Erfahrung bringen, wie der gerade ausgelastet ist. 

Ansonsten ist Alternative B auch in Ordnung. Es ist ja auch nicht mehr ewig Zeit…

Alles Gute

Philipp Schaer
